test/extended/openstack/kuryr.go

```diff
@@ -0,0 +1,77 @@
+package openstack
+
+import (
+	"context"
+
+	"github.com/gophercloud/gophercloud"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+)
+
+var _ = g.Describe("[sig-installer][Feature:openstack][Kuryr] Kuryr", func() {
+	var namespaces []*v1.Namespace
+	var networkClient *gophercloud.ServiceClient
+	var dc dynamic.Interface
+	var clientSet *kubernetes.Clientset
+
+	g.BeforeEach(func() {
+		g.By("preparing openshift dynamic client")
+		cfg, err := e2e.LoadConfig()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dc, err = dynamic.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		clientSet, err = e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		c, err := configv1client.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		network, err := c.ConfigV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		if network.Status.NetworkType != "Kuryr" {
+			e2eskipper.Skipf("Test can only be run with Kuryr SDN")
+		}
+
+		g.By("preparing openstack client")
+		networkClient, err = client(serviceNetwork)
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.AfterEach(func() {
+		clientSet, err := e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		g.By("Cleanup up Namespaces")
+		for _, ns := range namespaces {
```

Comment:
It seems there is no need to iterate over Namespaces and delete them at line 54, since that is already handled at `DeleteNamespaces`

Reply:
looks like this one still needs to be removed.

Reply:
@MaysaMacedo It should be gone now